src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceOpProcessorImpl.scala

```diff
@@ -28,17 +22,21 @@ private[tracker] class InstanceOpProcessorImpl(
     logger.debug(s"Process $op")
     val stateChange = stateOpResolver.resolve(op.op)
 
+    import akka.pattern.ask
+
+    import scala.concurrent.duration._
+
     stateChange.flatMap {
       case change: InstanceUpdateEffect.Expunge =>
         // Used for task termination or as a result from a UpdateStatus action.
         // The expunge is propagated to the instanceTracker which informs the sender about the success (see Ack).
-        repository.delete(change.instance.instanceId).map { _ =>
-          logger.debug(s"Expunged $change")
-          InstanceTrackerActor.Ack(op.sender, change)
-        }
-          .recoverWith(tryToRecover(op)(
-            expectedState = None, oldState = Some(change.instance), change.events))
-          .flatMap { ack: InstanceTrackerActor.Ack => notifyTaskTrackerActor(ack) }
+        implicit val taskTrackerQueryTimeout: Timeout = config.internalTaskTrackerRequestTimeout().milliseconds
+
+        val msg = InstanceTrackerActor.StateChanged(InstanceTrackerActor.Ack(op.sender, change))
+        logger.debug(s"Notify instance tracker actor: msg=$msg")
+        val f = (instanceTrackerRef ? msg).map(_ => ())
```

Comment:
So we just forward the message?

Reply:
This class is going to be removed anyway, right?